src/Castle.Windsor.Extensions.DependencyInjection/Scope/ExtensionContainerScope.cs

```diff
@@ -20,36 +20,40 @@ namespace Castle.Windsor.Extensions.DependencyInjection.Scope
 	using Castle.Core;
 	using Castle.MicroKernel;
 	using Castle.MicroKernel.Lifestyle.Scoped;
-	
+
 	internal class ExtensionContainerScope : ILifetimeScope, IDisposable
 	{
 		public static ExtensionContainerScope Current => current.Value;
 		public static string TransientMarker = "Transient";
 		protected static readonly AsyncLocal<ExtensionContainerScope> current = new AsyncLocal<ExtensionContainerScope>();
 		private readonly ExtensionContainerScope parent;
+		private readonly ExtensionContainerRootScope rootScope;
 		private readonly IScopeCache scopeCache;
 
-		protected ExtensionContainerScope(ExtensionContainerScope parent)
+		protected ExtensionContainerScope(
+			ExtensionContainerScope parent,
+			ExtensionContainerRootScope rootScope)
 		{
 			scopeCache = new ScopeCache();
-			if(parent == null)
-			{
-				this.parent = ExtensionContainerRootScope.RootScope;
-			}
-			else
-			{
-				this.parent = parent;
-			}
+			this.parent = parent ?? rootScope;
```

Comment:
Castle does not want c++ notations, must `if(parent!=null)`

Reply:
That is not true, C#'s null coalescing operator is perfectly acceptable. If Castle's coding style guide is confusing, we should update it.

Reply:
Good to know. My misunderstanding of https://github.com/castleproject/Home/blob/master/coding-standards.md#c-style-code